src/data/files.ts

```diff
@@ -0,0 +1,67 @@
+import { List } from 'immutable';
+
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+  ExcalidrawImageElement,
+  FileId,
+} from '@excalidraw/excalidraw/types/element/types';
+import { BinaryFileData, DataURL } from '@excalidraw/excalidraw/types/types';
+
+import { useAppDataContext } from '../components/context/AppDataContext';
+import { FileAppData } from '../config/appDataTypes';
+
+export const getListOfFileIds = (
+  elements: ExcalidrawImageElement[],
+): List<string | undefined> =>
+  List(elements).map((element) => element?.fileId as string);
+
+export const useFiles = (filesAppData: List<FileAppData>): BinaryFileData[] => {
+  const [files, setFiles] = useState<BinaryFileData[]>([]);
+  const { getFileContent } = useAppDataContext();
+  const getFile = useCallback(
+    async (fileAppData: FileAppData): Promise<BinaryFileData> => {
+      const { id, data: fileAppDataData } = fileAppData;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const reader = (file: any): Promise<DataURL> =>
+        new Promise((resolve, reject) => {
+          const fr = new FileReader();
+          fr.onload = () => resolve(fr.result as DataURL);
+          fr.onerror = (err) => reject(err);
+          fr.readAsDataURL(file);
+        });
+      const { name: fileName, mimetype } = fileAppDataData.extra.s3File;
+      const fileData = await getFileContent(id || '').then((result) =>
+        reader(result),
+      );
+      const file: BinaryFileData = {
+        mimeType: mimetype as BinaryFileData['mimeType'],
+        id: fileName as FileId,
```

Comment:
Wow this made me learn about type branding ! Super interesting, so I guess in this case, casting is ok !